### PR TITLE
feat: add on_chain_id to project table, create index

### PR DESCRIPTION
**Pull request by mhagel**
## Description

Needed for: regen-network/regen-registry#994

- add on_chain_id to project table, create index
---

### Author Checklist

*All items are required. Please add a note to the item if the item is not applicable and
please add links to any relevant follow up issues.*

I have...

- [x] targeted `dev` branch
- [x] provided a link to the relevant issue or specification
- [x] reviewed "Files changed" and left comments if necessary
- [x] confirmed all CI checks have passed
- [ ] submit a PR against `master` branch after this one (and other PRs depending on this one, e.g. on regen-network/regen-web, if applicable) get merged

### Reviewers Checklist

*All items are required. Please add a note if the item is not applicable and please **add
your handle next to the items reviewed if you only reviewed selected items**.*

I have...

- [ ] confirmed all author checklist items have been addressed
- [ ] reviewed code correctness and readability
- [ ] reviewed documentation is accurate
- [ ] reviewed tests
- [ ] manually tested (if applicable)

### CODE REVIEW

File: sql/V2_15__add_on_chain_id_to_project.sql [hunk 1]
```diff
@@ -0,0 +1,3 @@
+ALTER TABLE project
```
Comment (wgwz) [inline sql/V2_15__add_on_chain_id_to_project.sql:1]: nit: can you make this “add column if not exists”

File: sql/V2_15__add_on_chain_id_to_project.sql [hunk 2]
```diff
@@ -0,0 +1,3 @@
...
+    ADD COLUMN on_chain_id text unique;
+CREATE INDEX ON project ("on_chain_id");
```
Comment (wgwz) [inline sql/V2_15__add_on_chain_id_to_project.sql:3]: similarly, it’d be nice to have this as “create index if not exists”

**Review by wgwz — APPROVED** [review]
LGTM pending nits :-)

File: sql/V2_15__add_on_chain_id_to_project.sql (hunk 2 above)
Comment (mhagel) [reply sql/V2_15__add_on_chain_id_to_project.sql:3]: @wgwz Do you think there should also be a foreign key here?

**Review by wgwz — APPROVED** [review]
@mhagel sorry, i thought i had approved this last week, didn't go through i guess. thanks for handling the nits!

**Review by haveanicedavid — APPROVED** [review]

**Pull request merged**